packages/web/src/components/molecules/DeleteModal/DeleteModal.tsx

```diff
@@ -0,0 +1,103 @@
+import {useState} from 'react';
+
+import {Space} from 'antd';
+
+import {Button, FullWidthSpace, Text} from '@custom-antd';
+
+import {useDashboardNavigate} from '@hooks/useDashboardNavigate';
+import usePressEnter from '@hooks/usePressEnter';
+
+import Colors from '@styles/Colors';
+
+import {displayDefaultNotificationFlow} from '@utils/notification';
+
+import {notificationCall} from '..';
+
+import {DeleteModalWrapper} from './DeleteModal.styled';
+
+interface DeleteModalProps {
+  onDelete: (id: string) => Promise<any>;
+  onClose: () => void;
+  onConfirm?: () => void;
+  content?: React.ReactNode;
+  contentSize?: number;
+  title?: React.ReactNode;
+  subtitle?: React.ReactNode;
+  actionText?: string;
+  actionDisabled?: boolean;
+  defaultStackRoute?: string;
+  successMessage: string;
+  deleteOnEnter?: boolean;
+  idToDelete: string;
+}
+
+const DeleteModal: React.FC<DeleteModalProps> = ({
+  onClose,
+  onDelete,
+  onConfirm,
+  content,
+  contentSize,
+  title,
+  subtitle,
+  actionText,
+  actionDisabled,
+  defaultStackRoute,
+  successMessage,
+  deleteOnEnter = false,
+  idToDelete,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const back = useDashboardNavigate(defaultStackRoute || '');
+  const onEvent = usePressEnter();
+
+  const onDeleteCallback = async () => {
+    setIsLoading(true);
+    onDelete(idToDelete)
+      .then(displayDefaultNotificationFlow)
+      .then(() => {
+        notificationCall('passed', successMessage);
+        onClose();
+        if (onConfirm) {
+          onConfirm();
+        } else if (defaultStackRoute) {
+          back();
+        }
+      })
+      .catch(error => {
+        notificationCall('failed', error.title, error.message);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+  return (
+    <DeleteModalWrapper>
+      {content || (
+        <FullWidthSpace
+          size={contentSize || 16}
+          direction="vertical"
+          onKeyPress={event => {
+            onEvent(event, onDeleteCallback);
+          }}
```

Comment:
Given there is no input - should we just allow Enter key for deletion?

Reply:
adjusted to check for `deleteOnEnter` which is `false` by default.